sql/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java

```diff
@@ -83,12 +86,54 @@ public AnalyzedRelation analyze(Node node, Analysis analysis) {
 
     @Override
     protected AnalyzedRelation visitQuery(Query node, StatementAnalysisContext context) {
+        // In case of Set Operation
+        if (!node.getOrderBy().isEmpty() || node.getLimit().isPresent() || node.getOffset().isPresent()) {
+            // In the Future this can be a TwoRelationsSetOperation
```

Comment:
Not sure if the future-driven comment here makes sense unless you actually explain why and how so I'd rather remove it